### PR TITLE
api: add migration to set last_login property to created_at

### PR DESCRIPTION
**Pull request by henrybarreto**

### CODE REVIEW

File: api/store/mongo/migrations/migration_38.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+				{
+					{"$match", bson.D{
+						{"created_at", zeroTime},
+					}},
+				},
+				{
+					{"$replaceRoot", bson.D{{"newRoot", bson.M{"$mergeObjects": bson.A{"$$ROOT", bson.M{"created_at": "$last_login"}}}}}},
+				},
+				{
+					{"$out", "users"},
```
Comment (henrybarreto) [inline api/store/mongo/migrations/migration_38.go:34]: This line terminate the mongo's pipeline saying to write modified document. However, there is another staged, `$merge`, that could be used in the same way, in this case.

https://docs.mongodb.com/manual/reference/operator/aggregation/out/#mongodb-pipeline-pipe.-out

File: api/store/mongo/migrations/migration_38_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+	migrations := GenerateMigrations()[37:38]
+
+	timeNow := time.Now().UTC()
+	convertDate := func(t time.Time) string {
+		return t.Format("2006-01-02 15:04:05.000")
+	}
+
+	userNoCreated := models.User{
+		ID:         "userNoCreatedID",
```
Comment (otavio) [inline api/store/mongo/migrations/migration_38_test.go:38]: `userNoCreatedAt` in all bellow cases; also the other example would be `userWithCreatedAt`.

File: api/store/mongo/migrations/migration_38_test.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+		UserPassword: models.UserPassword{
+			Password: "",
+		},
+	}
+	userWithCreated := models.User{
+		ID:         "userWithCreatedID",
+		Namespaces: 0,
+		Confirmed:  false,
+		CreatedAt:  timeNow,
+		LastLogin:  timeNow,
```
Comment (otavio) [inline api/store/mongo/migrations/migration_38_test.go:57]: This case should use a date in the past so your test ensures it is not changing it.

**Review by otavio — CHANGES_REQUESTED** [review]

File: api/store/mongo/migrations/migration_38_test.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+			description: "Executes migration when user's created_at property is empty",
+			Test: func(t *testing.T) {
+				t.Helper()
+
+				var userMigrated *models.User
+				err = db.Client().Database("test").Collection("users").FindOne(context.TODO(), bson.D{{"username", userNoCreatedAt.Username}}).Decode(&userMigrated)
+				assert.NoError(t, err)
+				assert.Equal(t,
+					Expected{CreatedAt: convertDate(userNoCreatedAt.LastLogin), LastLogin: convertDate(userNoCreatedAt.LastLogin)},
+					Expected{CreatedAt: convertDate(userMigrated.CreatedAt), LastLogin: convertDate(userMigrated.LastLogin)},
```
Comment (otavio) [inline api/store/mongo/migrations/migration_38_test.go:91]: What happens if user didn't login yet?

**Review by otavio — CHANGES_REQUESTED** [review]

File: api/store/mongo/migrations/migration_38_test.go (hunk 4 above)
Comment (henrybarreto) [reply api/store/mongo/migrations/migration_38_test.go:91]: Theoretically, if time is "zero"for both, it copies the zero. 

I'm working on a better query for the aggregation.

File: api/store/mongo/migrations/migration_38.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+		}).Info("Applying migration")
+		zeroTime := time.Time{}.UTC()
+		_, err := db.Collection("users").Aggregate(context.TODO(),
+			mongo.Pipeline{
+				{
+					{"$match", bson.D{
+						{"$and", bson.A{
+							bson.M{"created_at": bson.M{"$eq": zeroTime}},
+							bson.M{"last_login": bson.M{"$ne": zeroTime}},
+						}},
```
Comment (henrybarreto) [inline api/store/mongo/migrations/migration_38.go:30]: That seems preciser

**Pull request merged**